src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -244,11 +247,20 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
       fetchData(cursors).map { sr =>
         val items = sr.items
         val nextCursor = sr.nextCursor
-        val shouldStopEarly = items.isEmpty || (applyTimestampTermination && items.last.lastUpdatedTime >=
-          terminationTimeStamp)
+        val shouldStopEarly = items.nonEmpty && applyTimestampTermination && items.last.lastUpdatedTime >= terminationTimeStamp
         val next =
-          (nextCursor, shouldStopEarly) match {
-            case (Some(cursor), false) =>
+          (nextCursor, items.isEmpty, shouldStopEarly) match {
+            case (Some(cursor), true, _) =>
+              if (syncCursorSaveCallbackUrl.isDefined && jobId.isDefined && cursorName.isDefined) {
```

Comment:
for little more fancy unpacking of several options might also do
```scala
(syncCursorCallbackUrl, jobId, cursorName) match {
  case (Some(syncCursorCallbackUrl), Some(jobId), Some(cursorName)) =>
}
```
or better yet create a small wrapper class and constructor/factory method for it
```scala
case class SyncCursorCallbackSettings(callbackUrl: String, jobId: String, cursorName: String)
```
about so

Reply:
The downside with the match is that it has to be complete, so you need a "case => " at the end. If not you will end up with scala.MatchError . This is why some don't use match when you have a single case. In some cases, such as transform in Spark, there is an implicit default case, so there it isn't a problem, but for normal code, this is not the case.

Reply:
yep, it'd need a default case but upside is not calling partial `.get` functions (ideally there should be a linter to check if .get is guarded by emptiness check as scala compiler won't do that)

There's also stuff like `PartialFunction` but probably won't save much. Or quite specialized `Apply[Option].map3`
```scala
              Apply[Option].map3(syncCursorSaveCallbackUrl, jobId, cursorName){
                case (syncCursorSaveCallbackUrl, jobId, cursorName) => {
                  
                }
              }
```
also a bit too complex